OpenPGL.NET/Field.cs

```diff
@@ -200,15 +210,51 @@ public class VMMDirectionalSettings : DirectionalSettings {
     }
 }
 
+public enum DQTLeafEstimator {
+    RejectionSampling = 0,
+    PerLeaf
+};
+
+public enum DQTSplitMetric {
+    Mean = 0,
+    SecondMoment
+};
+
+public class DQTDirectionalSettings : DirectionalSettings {
+    public DQTLeafEstimator LeafEstimator = DQTLeafEstimator.RejectionSampling;
+    public DQTSplitMetric SplitMetric = DQTSplitMetric.Mean;
+    public float SplitThreshold = 0.01f;
+    public float FootprintFactor = 1;
+    public int MaxLevels = 12;
+    
+    internal override OpenPGL.PGL_DIRECTIONAL_DISTRIBUTION_TYPE GetDistType() {
+        return OpenPGL.PGL_DIRECTIONAL_DISTRIBUTION_TYPE.PGL_DIRECTIONAL_DISTRIBUTION_QUADTREE;
+    }
+
+    internal override void SetArguments(ref OpenPGL.PGLFieldArguments target) {
+        target.directionalDistributionType = GetDistType();
+
+        OpenPGL.PGLDQTFactoryArguments args = new() {
+            leafEstimator = (OpenPGL.PGLDQTLeafEstimator) LeafEstimator,
+            splitMetric = (OpenPGL.PGLDQTSplitMetric) SplitMetric,
+            splitThreshold = SplitThreshold,
+            footprintFactor = FootprintFactor,
+            maxLevels = (uint)int.Max(0, MaxLevels),
+        };
+        Marshal.StructureToPtr<OpenPGL.PGLDQTFactoryArguments>(args, target.directionalDistributionArguments, true);
+    }
+}
+
 public struct FieldSettings {
     public SpatialSettings SpatialSettings { get; init; }
     public DirectionalSettings DirectionalSettings { get; init; }
 
     internal OpenPGL.PGLFieldArguments MakeArguments() {
+        // We require the correct struct and dist type for preallocation done inside pglFieldArgumentsSetDefaults

```

Comment:
```suggestion
       
```

Reply:
Since the memory is freed by device.NewField, probably best to move this logic to the constructor of field in the wrapper like
```csharp
public Field(FieldSettings settings = new()) {
        device = new();
        
        // pglFieldArgumentsSetDefaults allocates memory for the settings, that we can then write to.
        // pglDeviceNewField frees that memory again
        OpenPGL.pglFieldArgumentsSetDefaults(out var arguments, 
            settings.SpatialSettings.StructType, settings.DirectionalSettings.DistType);
        SpatialSettings.SetArguments(ref arguments);
        DirectionalSettings.SetArguments(ref arguments);

        Handle = OpenPGL.pglDeviceNewField(device.Ptr, arguments);
        Debug.Assert(Handle != nint.Zero);
    }

```

Reply:
Yeah, thats a better approach